### PR TITLE
insertion sort, blog, testing

### PR DESCRIPTION
**Pull request by shaybrow**

### CODE REVIEW

**Comment by shaybrow** [comment]
- [ ] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: challenges/lib/src/main/java/challenges/utilities/InsertionSort.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            while (j>= 0 && input[j] > temp){
+                input[j+1] = input[j];
+                j = j-1;
+            }
+            input[j+1] = temp;
+            printArray(input);
+        }
+
+    }
+//    borrowed this from https://www.geeksforgeeks.org/insertion-sort/
```
Comment (Mattpet26) [inline challenges/lib/src/main/java/challenges/utilities/InsertionSort.java:19]: Good work giving them credit, it's great to see this.

File: challenges/lib/src/test/java/challenges/InsertionSortTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import challenges.utilities.InsertionSort;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class InsertionSortTest {
+
+    @Test
```
Comment (Mattpet26) [inline challenges/lib/src/test/java/challenges/InsertionSortTest.java:11]: Good work on your tests. Utilizing a negative number to see if that'd break the sort, nicely done!
